src/track/globaltrackcache.cpp

```diff
@@ -580,6 +585,18 @@ TrackRef GlobalTrackCache::initTrackId(
     return trackRefWithId;
 }
 
+void GlobalTrackCache::purgeTrackId(TrackId trackId) {
+    DEBUG_ASSERT(trackId.isValid());
+
+    const auto trackById(m_tracksById.find(trackId));
+    if (m_tracksById.end() != trackById) {
+        Track* track = trackById->second->getPlainPtr();
+        track->resetId();
+        track->setDateAdded(QDateTime());
```

Comment:
Resetting the date does definitely not belong here, completely wrong context and abstraction level.

Reply:
It is moved here, because normally we have no track object when purging a track. It is a database only action that operates with track IDs only. Only at this point if we have found the ID in the cache, we have access to the track object. 

We can solve the abstraction issue by instantiate a track object before purging it, but that is just extra CPU and DB reads. Alternative, we may not reset The date added and remove the debug assert expecting a NULL date before adding. 

What do you think?

Reply:
Then let's keep the date unmodified instead of violating abstraction rules. The Track object needs to be aware of the use case that it might be re-added. Please log a message on level INFO if that happens.